### PR TITLE
Adding nonEmpty pipe to unwrap option values

### PR DESCRIPTION
**Pull request by yilinwei**

### CODE REVIEW

File: core/shared/src/main/scala/fs2/pipe.scala [hunk 1]
```diff
@@ -336,6 +336,10 @@ object pipe {
       }
     }
 
+  /** Filters any 'None'. */
+  def nonEmpty[F[_], I]: Pipe[F, Option[I], I] =
```
Comment (mpilquist) [inline core/shared/src/main/scala/fs2/pipe.scala]: Name wise, what do you think of `unNone` for symmetry with `unNoneTerminate`?

**Comment by yilinwei** [comment]
@mpilquist Done.

File: core/shared/src/main/scala/fs2/pipe.scala [hunk 2]
```diff
@@ -458,6 +458,9 @@ object pipe {
   def unchunk[F[_],I]: Pipe[F,I,I] =
     _ repeatPull { _.receive1 { case (i, h) => Pull.output1(i) as h }}
 
+  /** Filters any 'None'. */
+  def unNone[F[_], I]: Pipe[F, Option[I], I] = _.filter(_.isDefined).map(_.get)
```
Comment (mpilquist) [inline core/shared/src/main/scala/fs2/pipe.scala]: Sorry, 1 more comment :)

This would be more efficient as `_.collect { case Some(i) => i }`

**Comment by yilinwei** [comment]
@mpilquist Changed.

**Pull request merged**